### PR TITLE
Jeanette/scoot154 watch job command

### PR DESCRIPTION
**Pull request by JeanetteBruno**
Watch_Job: changes mainly to scootapi and swarmtest: added watch_job to scootapi, returning runid from run_job and adding -setup_then_wait option to swarmtest.
corresponds with https://phabricator.twitter.biz/D8760

### CODE REVIEW

File: runner/execer/os/os_execer.go [hunk 1]
```diff
@@ -2,11 +2,10 @@ package os
 
 import (
 	"errors"
+	"github.com/scootdev/scoot/runner/execer"
```
Comment (dbentley) [inline runner/execer/os/os_execer.go]: We've been segregating imports. So stdlib are in one block, then scoot in another.

Is there a reason to change?

File: runner/local/output.go [hunk 2]
```diff
@@ -61,6 +61,7 @@ func (o *localOutput) Close() error {
 // directly to a new process's stdout/stderr.)
 // We care about this cleverness, so Output both is-a and has-a Writer
 func (o *localOutput) WriterDelegate() io.Writer {
+
```
Comment (dbentley) [inline runner/local/output.go]: extra line?

File: sched/scheduler/stateful_scheduler.go [hunk 3]
```diff
@@ -9,6 +9,7 @@ import (
 	"github.com/scootdev/scoot/saga"
 	"github.com/scootdev/scoot/sched"
 	"github.com/scootdev/scoot/sched/worker"
+	"strings"
```
Comment (dbentley) [inline sched/scheduler/stateful_scheduler.go]: I'd put this in a new block above

File: scootapi/client/get_status_cmd.go [hunk 4]
```diff
@@ -19,6 +20,11 @@ func (c *getStatusCmd) registerFlags() *cobra.Command {
 func (c *getStatusCmd) run(cl *Client, cmd *cobra.Command, args []string) error {
 
 	log.Println("Checking Status for Scoot Job", args)
+
+	if args == nil || len(args) == 0 {
```
Comment (dbentley) [inline scootapi/client/get_status_cmd.go]: I think len(args) returns 0 if args == nil, so no need for a separate check (unlike java)

File: scootapi/client/run_job_cmd.go [hunk 5]
```diff
@@ -29,6 +29,10 @@ func (c *runJobCmd) registerFlags() *cobra.Command {
 func (c *runJobCmd) run(cl *Client, cmd *cobra.Command, args []string) error {
 	log.Println("Running on scoot", args)
 
+	if args == nil || len(args) == 0 {
+		return errors.New("a job id must be provided")
```
Comment (dbentley) [inline scootapi/client/run_job_cmd.go]: I think this is no longer true with Ryan's changes (which you've incorporated below)

File: scootapi/client/run_job_cmd.go [hunk 6]
```diff
@@ -65,5 +71,9 @@ func (c *runJobCmd) run(cl *Client, cmd *cobra.Command, args []string) error {
 			return fmt.Errorf("Error running job: %v %T", err, err)
 		}
 	}
+
+	log.Println(fmt.Printf("JobID:%s\n", jobId))
```
Comment (dbentley) [inline scootapi/client/run_job_cmd.go]: 1) fmt.Printf already prints to stdout, so you don't need to wrap it in a log.Println
2) for stdout, I'd suggest just printing the jobId
3) then *also* log.Println with more context

That will make stdout consumable by other programs, and stderr useful for humans. WDYT?

File: scootapi/client/watch_job_cmd.go [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+func (c *watchJobCmd) run(cl *Client, cmd *cobra.Command, args []string) error {
+
+	log.Println("Watching job:", args)
+
+	if args == nil || len(args) == 0 {
+		return errors.New("a job id must be provided")
+	}
+
+	client, err := cl.Dial()
+
```
Comment (dbentley) [inline scootapi/client/watch_job_cmd.go]: lots of extra lines between a statement and then "if err != nil" below it

Comment (JeanetteBruno) [inline scootapi/client/watch_job_cmd.go]: removed blank lines above err checks

File: scootapi/client/run_job_cmd.go (hunk 6 above)
Comment (JeanetteBruno) [inline scootapi/client/run_job_cmd.go]: done

File: runner/local/output.go (hunk 2 above)
Comment (JeanetteBruno) [inline runner/local/output.go]: removed

File: scootapi/client/get_status_cmd.go (hunk 4 above)
Comment (JeanetteBruno) [inline scootapi/client/get_status_cmd.go]: done - removed args == nil clause

File: scootapi/client/run_job_cmd.go (hunk 5 above)
Comment (JeanetteBruno) [inline scootapi/client/run_job_cmd.go]: removed

**Pull request merged**